crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -364,11 +367,11 @@ impl RoomEventCache {
     /// handlers?
     pub async fn subscribe(
         &self,
+        event_cache: &EventCache,
```

Comment:
I would prefer to pass `EventCache` to `RoomEventCache::new`: it would keep the `subscribe` API simple, and may be re-used somewhere else. Thoughts?

Reply:
Yeah, this part is the least pretty of the patch: the only reason we're passing the `EventCache` by parameter here is because we're making use of the `store` in this method, and I liked the invariant that only the `EventCache` had access to the store (not the `RoomEventCache`). Maybe it's a bad idea long term, though, and the lock could be an `Arc<Mutex<>>` instead. Thoughts?

Reply:
Even in my back-pagination work, I need the store in more places at the room level, so I'll move to an `Arc<Mutex<...>>` instead.